### ISSUE TITLE
Capturing app state with URLs

### ISSUE THREAD

**Issue by Robinlovelace** [nptscot/nptscot.github.io#266]
Currently the app captures only changes to high-level layers being on and off.

Would it be possible to capture more info like location of the sliders, which sub-layers (e.g. infrastructure type), etc is on?

Question for @mvl22.

**Comment by mvl22** [comment]
Yes, one of the most recent commits will facilitate this as layer controls are now consistently tagged. Will work on it shortly as I find myself wanting this regularly.

**Comment by Robinlovelace** [comment]
Fantastic, thanks for the update  Martin and good luck with it. Lots of benefits when this is implemented.

**Comment by Robinlovelace** [comment]
Nudge on this @mvl22, first day back at work, this seems like the chunkiest issue in the frontend for the v2 milestone. Anything I can do to help? 

FYI had a good session with Zhao to finalise the v2 checklist on the backend side, we're getting there..

**Comment by Robinlovelace** [comment]
Hi @mvl22 with apologies for badgering, could provide an ETA for this? If there are any blockers or any way we can help let us know.

**Comment by mvl22** [comment]
Should be in place by Monday night. Hopefully you got my update e-mail a few days ago.

**Comment by Robinlovelace** [comment]
Thanks for the update Martin, anything we can do to help let us know.

**Comment by mvl22** [comment]
I've been moving the layers over to a proper state model. However, I noticed a regression in the rnet/rnet-simplified switching stuff, which is a bit of a pain and has slowed me down, so this won't be finalised tonight I'm afraid. Once I can clear that bug, the remaining bits should be straightforward. Will press on tomorrow, as I want this out my head ASAP.

Code is on the new `layer-updates` branch.

**Comment by Robinlovelace** [comment]
Great stuff thanks for the update Martin.

**Comment by mvl22** [comment]
Working through some unblocking commits.

First is #298, which decouples the accordion system from popups.

**Comment by mvl22** [comment]
Next is replacing the DOM-based state with a proper state object.

This is now implemented in #301; please review and merge.

**Comment by mvl22** [comment]
Parameters are now being tracked into the state in 500a8d0a82de5944a5da9fe97b38e8ec299d1374.

Now just need to wire this up to URLs and vice-versa.

**Comment by mvl22** [comment]
Ready for review, testing out in reality, and merge, in PR #336.

**Comment by Robinlovelace** [comment]
Amazing, thanks, will test properly tomorrow.

**Comment by Robinlovelace** [comment]
This is done.
